app/src/main/java/com/rage/clamber/Fragments/CommentsFragment.java

```diff
@@ -0,0 +1,187 @@
+import com.rage.clamber.Adapters.CommentsRecyclerViewAdapter;
+import com.rage.clamber.Data.Comment;
+import com.rage.clamber.Data.User;
+import com.rage.clamber.Networking.ApiManager;
+import com.rage.clamber.Networking.Requests.NewCommentRequest;
+import com.rage.clamber.R;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import butterknife.Bind;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+/**
+ * A fragment to hold and display user Comments for each climb.
+ */
+public class CommentsFragment extends Fragment {
+
+    public static final String TAG = CommentsFragment.class.getSimpleName();
+    protected User mainUser;
+    protected List<Comment> comments;
+    protected int climbId;
+    protected CommentsRecyclerViewAdapter adapter;
+
+    @Bind(R.id.comments_fragment_recycler_view)
+    RecyclerView recyclerView;
+    @Bind(R.id.comments_fragment_no_comments_text)
+    TextView noCommentsText;
+
+
+    public CommentsFragment() {
+        // Required empty public constructor
+    }
+
+    public static CommentsFragment newInstance(User user, int climb_id) {
+        Bundle args = new Bundle();
+        args.putParcelable(HomePage.ARG_USER, user);
+        args.putInt(ClimbsRecyclerViewAdapter.ARG_CLIMB, climb_id);
+        CommentsFragment fragment = new CommentsFragment();
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+
+        View rootView = inflater.inflate(R.layout.fragment_comments, container, false);
+        ButterKnife.bind(this, rootView);
+
+        mainUser = getArguments().getParcelable(HomePage.ARG_USER);
+        climbId = getArguments().getInt(ClimbsRecyclerViewAdapter.ARG_CLIMB);
+        comments = new ArrayList<>();
+        recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
+        adapter = new CommentsRecyclerViewAdapter(comments, mainUser, getContext());
+        recyclerView.setAdapter(adapter);
+
+        getCommentsByClimb(climbId);
+
+        return rootView;
+    }
+
+    /**
+     * Method to make a network call to the Clamber Server to get all Comments by Climb Id. The
+     * climbsRecyclerViewAdapter has an OnClick method on the comments button that launches this
+     * fragment and triggers this network call to display comments.
+     *
+     * @param climbId - id of the climb that was selected.
+     */
+    public void getCommentsByClimb(int climbId) {
+        ConnectivityManager connectivityManager = (ConnectivityManager) getActivity().getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
+        if (networkInfo != null && networkInfo.isConnected()) {
```

Comment:
I wonder, does Retrofit handle the checking for connectivity for you? Seems like the kind of thing it might do.